src/main/java/agents/leeroy/Leeroy.java

```diff
@@ -67,16 +90,106 @@ private void setPhase(Risk game) {
+                    .collect(Collectors.toList());
+            for (int i = territoriesRelations.size() - 1; i == 0; i--) {
+                if (1d - territoriesRelations.get(i).getValue() > TROOPS_RELATION_THRESHOLD) {
+                    return RiskAction.select(territoriesRelations.get(i).getKey());
+                }
+            }
+            log.warn("Did not find fitting territory in selected continent");
+            return Util.selectRandom(game.getPossibleActions());
+        } else {
+            return board
+                    .getTerritories()
+                    .entrySet()
+                    .stream()
+                    .filter(integerRiskTerritoryEntry -> integerRiskTerritoryEntry.getValue().getOccupantPlayerId() == game.getCurrentPlayer())
+                    .map(integerRiskTerritoryEntry -> {
+                        var enemyTroops = board.neighboringEnemyTerritories(integerRiskTerritoryEntry.getKey())
+                                .stream()
+                                .reduce(Integer::sum)
+                                .map(Integer::doubleValue);
+                        return enemyTroops.
+                                map(noTroops -> new AbstractMap.SimpleImmutableEntry<>(integerRiskTerritoryEntry.getKey(), integerRiskTerritoryEntry.getValue().getTroops() / noTroops))
+                                .orElseGet(() -> new AbstractMap.SimpleImmutableEntry<>(integerRiskTerritoryEntry.getKey(), 1d));
+                    })
+                    .min(Comparator.comparingDouble(AbstractMap.SimpleImmutableEntry::getValue))
+                    .map(integerDoubleSimpleImmutableEntry -> RiskAction.select(integerDoubleSimpleImmutableEntry.getKey()))
+                    .orElse(Util.selectRandom(game.getPossibleActions()));
+        }
+    }
+
+    private Node searchBestNode(Node node, Function<Node, Node> expansionFunction, Function<Node, Integer> evaluationFunction) {
+        performMCTS(node, expansionFunction, evaluationFunction);
+        return node.getSuccessors().stream().max(Comparator.comparingDouble(Node::getWinScore)).get();
     }
 
     private RiskAction attackTerritory(Risk risk) {
         // TODO: MCTS
-        return Util.selectRandom(AttackActionSupplier.createActions(risk));
+        Set<RiskAction> attackActions = AttackActionSupplier.createActions(risk);
+        Optional<RiskAction> highAtkAction = attackActions
+                .stream()
+                .sorted(Comparator.comparingInt(action -> action.troops() *-1))
+                .findFirst();
+        return highAtkAction.orElse(RiskAction.endPhase());
+    }
+
+    private RiskAction fortifyTerritory(Risk risk) {
+        // TODO: MCTS
+        Set<RiskAction> fortificationActions = FortificationActionSupplier.createActions(risk);
+        Optional<RiskAction> bestAction = fortificationActions
+                .stream()
+                .sorted(Comparator.comparingInt(action -> action.troops() *-1))
```

Comment:
I think you could also do this with simply `.min(Comparator.comparingInt())`, no?

Reply:
sure, thanks :-)